contrib/manyclangs-setup.sh

```diff
@@ -0,0 +1,208 @@
+ELFSHAKER_ARCH=${ELFSHAKER_ARCH-$MANYCLANGS_ARCH}
+
+# If modifying this, beware of the contrib scripts.
+MANYCLANGS_CLANG_VERSION=${MANYCLANGS_CLANG_VERSION-12}
+
+# Installation path of elfshaker. Installed if not present.
+ELFSHAKER_BIN=${ELFSHAKER_BIN-$HOME/.local/bin/elfshaker}
+ELFSHAKER_BIN_DIR=$(dirname "${ELFSHAKER_BIN}")
+mkdir -p "${ELFSHAKER_BIN_DIR}"
+
+export PATH="${ELFSHAKER_BIN_DIR}:$PATH"
+
+install_system_dependencies() {
+    if { 
+        command -v c++ &&
+        command -v cc &&
+        command -v clang-${MANYCLANGS_CLANG_VERSION} &&
+        command -v cmake &&
+        command -v git &&
+        command -v go &&
+        command -v jq &&
+        command -v lld-${MANYCLANGS_CLANG_VERSION} &&
+        command -v make &&
+        command -v ninja &&
+        command -v pv &&
+        command -v python3 &&
+        :
+    } > /dev/null
+    then
+        return
+    fi
+    PKGS=(
+        clang-${MANYCLANGS_CLANG_VERSION}
+        cmake
+        g++
+        gcc
+        git
+        golang-go
+        jq
+        lld-${MANYCLANGS_CLANG_VERSION}
+        make
+        ninja-build
+        pv
+        python3
+    )
+    sudo apt update
+    sudo apt install -y "${PKGS[@]}"
+}
+
+fetch_elfshaker_source() {
+    if [ ! -d elfshaker ]
+    then
+        git clone https://github.com/elfshaker/elfshaker
+    fi
+}
+
+install_elfshaker_build_dependencies() {
+    if command -v elfshaker > /dev/null
+    then
+        return
+    fi
+
+    if ! command -v cargo &>/dev/null
+    then
+        echo "Rust unavailable, installing"
+        echo
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -- -y
+    fi
+
+    if ! command -v cc &> /dev/null
+    then
+        sudo apt install -y gcc
+    fi
+}
+
+build_elfshaker() {
+    (cd elfshaker && cargo build --release && cp target/release/elfshaker "${ELFSHAKER_BIN}")
+}
+
+install_elfshaker() {
+    if [ "$ELFSHAKER_FROM_SOURCE" ]
+    then
+        install_elfshaker_build_dependencies
+        build_elfshaker
+    else
+        ELFSHAKER_TGZ=elfshaker_${ELFSHAKER_BINARY_VERSION}_${ELFSHAKER_ARCH}-unknown-linux-musl.tar.gz
+        wget --continue \
+            https://github.com/elfshaker/elfshaker/releases/download/${ELFSHAKER_BINARY_VERSION}/${ELFSHAKER_TGZ} \
+            https://github.com/elfshaker/elfshaker/releases/download/${ELFSHAKER_BINARY_VERSION}/${ELFSHAKER_TGZ}.sha256sum
+        sha256sum --check ${ELFSHAKER_TGZ}.sha256sum
+        
+        tar --extract --directory $HOME/.local/bin --file ${ELFSHAKER_TGZ} --strip-components=1 elfshaker/elfshaker
+    fi
+}
+
+fetch_llvm() {
+    if [ ! -d llvm-project ]
+    then
+        git clone --no-checkout https://github.com/llvm/llvm-project llvm-project
+    fi
+    (cd llvm-project && git fetch origin)
+}
+
+install_jobserver() {
+    if ! command jobserver &> /dev/null
+    then
+        if [ ! -d jobclient ]
+        then
+            git clone https://github.com/olsner/jobclient
```

Comment:
Is it a good idea to checkout a specific commit here, for security reasons?

Reply:
Yes, let's do that.